core/src/test/java/de/evoila/cf/security/utils/RandomStringTest.java

```diff
@@ -0,0 +1,118 @@
+package de.evoila.cf.security.utils;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class RandomStringTest {
+
+    @Mock
+    private Random mockRandom;
+
+    @Test
+    void RandomString_onlyLowerCase() {
+        when(mockRandom.nextInt(52))
+                .thenAnswer(new Answer() {
+                                int index = 0;
+                                @Override
+                                public Object answer(InvocationOnMock invocationOnMock) {
+                                    if (index == 52) {
+                                        index = 0;
+                                    }
+                                    return index++;
+                                }
+                            }
+                );
+        RandomString randomString = new RandomString(104, mockRandom, false,true);
+        String string = randomString.nextString();
+        assertEquals(string.length(), 104);
+        assertEquals("abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz", string);
+    }
+
+    @Test
+    void RandomString_LowerAndUpperCase() {
+        when(mockRandom.nextInt(52))
+                .thenAnswer(new Answer() {
+                                int index = 0;
+                                @Override
+                                public Object answer(InvocationOnMock invocationOnMock) {
+                                    if (index == 52) {
+                                        index = 0;
+                                    }
+                                    return index++;
+                                }
+                            }
+                );
+        RandomString randomString = new RandomString(52, mockRandom, false,false);
+        String string = randomString.nextString();
+        assertEquals(string.length(), 52);
+        assertEquals("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz", string);
+
+    }
+
+    @Test
+    void RandomString_LowerCaseAndNumbers() {
```

Comment:
Please do not use underscores in method names. In Java the camel case notation is commonly used for naming.